File: part-2/module-5-multilingual-design/workspace_folder/src/example_pkg_py/example_pkg_py/testpub.py
#Daniel Yanke
#10/3/2024

import rclpy
from rclpy.node import Node

from example_test_interfaces.msg import ExampleTest

class PyPublisher(Node):
    def __init__(self):
        super().__init__("py_publisher")
        self.publisher = self.create_publisher(ExampleTest, "test_topic", 10)
        self.count = 0
        self.timer = self.create_timer(0.5, self.timer_callback)
    def timer_callback(self):
        msg = ExampleTest()
        msg.unity_id = "djyanke"
        msg.seq_number = self.count
        self.count += 1
        msg.language = "Python"
        self.publisher.publish(msg)
        self.get_logger().info(f"Publishing: '{msg.unity_id}', '{msg.seq_number}'")

def main(args = None):
    rclpy.init(args = args)
    python_publisher = PyPublisher()
    rclpy.spin(python_publisher)
    python_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: part-2/module-5-multilingual-design/workspace_folder/src/example_pkg_cpp/src/testpub.cpp
/*
Author: Daniel Yanke
Date: 10/3/2024
*/

#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "example_test_interfaces/msg/example_test.hpp"

class CppPublisher : public rclcpp::Node{
public:
    CppPublisher() : Node("cpp_publisher"), count_(0){
        publisher_ = this->create_publisher<example_test_interfaces::msg::ExampleTest>("test_topic", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&CppPublisher::timer_callback, this));
    }
private:
    void timer_callback(){
        auto message = example_test_interfaces::msg::ExampleTest();
        message.unity_id = "djyanke";
        message.seq_number = count_++;
        message.language = "C++";
        RCLCPP_INFO(this->get_logger(), "Publishing: '%s', '%s'", message.unity_id.c_str(), std::to_string(message.seq_number).c_str());
        publisher_->publish(message);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<example_test_interfaces::msg::ExampleTest>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CppPublisher>());
    rclcpp::shutdown();
    return 0;
}